src/client/public/scripts/components/base-element.ts

```diff
@@ -1,17 +1,44 @@
 import {render} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
 import {TemplateResult} from '../../../../../node_modules/lit-html/lit-html.js';
 
+type Debouncer = (task: Function) => Promise<void>;
+
+function createDebouncer(): Debouncer {
+  let queued = false;
+  return async (task: Function) => {
+    if (queued) {
+      return;
+    }
+    queued = true;
+    await Promise.resolve();
```

Comment:
Can you add some comments on why you are await a promise.resolve - I'm assuming to allow microtasks to finish before rendering.

Is there a reason this is preferable over rAF?

Reply:
In the microtask case you are avoiding rerender after every render call in a single turn. The raf case also works as long as you are okay with waiting the extra time to render. Microtask is just the minimum viable timing to use for this model IMO.

Reply:
What he said.